chips/earlgrey/src/chip.rs

```diff
@@ -48,28 +100,11 @@ impl<A: 'static + Alarm<'static>> EarlGrey<A> {
 
     unsafe fn handle_plic_interrupts(&self) {
         while let Some(interrupt) = plic::next_pending() {
-            match interrupt {
-                interrupts::UART_TX_WATERMARK..=interrupts::UART_RX_PARITY_ERR => {
-                    uart::UART0.handle_interrupt()
-                }
-                int_pin @ interrupts::GPIO_PIN0..=interrupts::GPIO_PIN31 => {
-                    let pin = &gpio::PORT[(int_pin - interrupts::GPIO_PIN0) as usize];
-                    pin.handle_interrupt();
-                }
-                interrupts::HMAC_HMAC_DONE..=interrupts::HMAC_HMAC_ERR => {
-                    hmac::HMAC.handle_interrupt()
-                }
-                interrupts::USBDEV_PKT_RECEIVED..=interrupts::USBDEV_CONNECTED => {
-                    usbdev::USB.handle_interrupt()
-                }
-                interrupts::PWRMGRWAKEUP => {
-                    pwrmgr::PWRMGR.handle_interrupt();
-                    self.check_until_true_or_interrupt(
-                        || pwrmgr::PWRMGR.check_clock_propagation(),
-                        None,
-                    );
-                }
-                _ => debug!("Pidx {}", interrupt),
+            if interrupt == interrupts::PWRMGRWAKEUP {
+                self.pwrmgr.handle_interrupt();
+                self.check_until_true_or_interrupt(|| self.pwrmgr.check_clock_propagation(), None);
+            } else if !self.plic_interrupt_service.service_interrupt(interrupt) {
+                debug!("Pidx {}", interrupt);
```

Comment:
Why do we have the two levels of interrupts?

Reply:
The way the chip implementation works now, power manager interrupts must be handled for the chip to work at all, so it isn't necessary for the power manager to be part of the `DefaultPeripherals` struct because any reimplementation would *have* to include it in order to use the other chip code. Also, calling `check_until_true_or_interrupt()` requires a reference to the Chip, and I didn't want to create a circular dependency here.